crates/cli/commands/src/common.rs

```diff
@@ -65,11 +65,16 @@ impl EnvironmentArgs {
         }
 
         let config_path = self.config.clone().unwrap_or_else(|| data_dir.config());
-        let mut config: Config = confy::load_path(config_path)
-            .inspect_err(
-                |err| warn!(target: "reth::cli", %err, "Failed to load config file, using default"),
+        let mut config = if config_path.exists() {
+            toml::de::from_str(
+                &std::fs::read_to_string(&config_path)
+                    .map_err(|e| warn!("Failed to read config file: {}", e))
+                    .unwrap_or_default(),
             )
-            .unwrap_or_default();
+            .unwrap_or_default()
```

Comment:
we should inspect and log the error in the same way we did before

Reply:
Just fixed this with a two stage inspection: one for reading the file and one for parsing for a better granularity of information, let me know if this is ok